packages/react/popper/src/Popper.tsx

```diff
@@ -279,8 +279,8 @@ const PopperContent = React.forwardRef<PopperContentElement, PopperContentProps>
           position: strategy,
           left: 0,
           top: 0,
-          transform: isPlaced
-            ? `translate3d(${Math.round(x)}px, ${Math.round(y)}px, 0)`
+          transform: isPositioned
+            ? `translate3d(${Math.round(x as number)}px, ${Math.round(y as number)}px, 0)`
```

Comment:
Checking `isPositioned` making sure that `x` and `y` are not `null`.

Reply:
You can use the new `floatingStyles` object returned from the hook `v2` instead of computing the styles here. It performs rounding based on DPR and should be more accurate.

This point: `// keep off the page when measuring` shouldn't matter because the browser doesn't paint until it's positioned anyway.

Reply:
> the browser doesn't paint until it's positioned anyway

@atomiks Why not? I tried to remove this and in heavily loaded story, the difference is noticable.


https://github.com/radix-ui/primitives/assets/10157660/beb1d296-fe6a-4598-9444-3e9da5d0d1cf

Reply:
I see! Yeah, for SSR etc, then keeping it out of view (or using `visibility: isPositioned ? undefined : 'hidden'`) is the correct approach.

Reply:
I just noticed that there is `translate` in `floatingStyles` and `translate3d` in Radix. I will change it to `translate` to make it consistent.

Reply:
Yeah, `willChange: 'transform` makes `translate` act as `translate3d`. It's only applied to high PPI displays to reduce blurring issue on Windows

Reply:
<img width="551" alt="CleanShot 2023-05-17 at 13 42 08@2x" src="https://github.com/radix-ui/primitives/assets/1539897/23cb986e-7ffa-4fe9-a3a5-2b35f9f555ab">

> It performs rounding based on DPR and should be more accurate.

@atomiks I'm guessing this is what we're seeing here caught up by Chromatic?

Reply:
I'd assume so yes